models/jupiter/solana/jupiter_solana_perp_events.sql

```diff
@@ -0,0 +1,113 @@
+    , block_slot
+    , block_time
+    , cast(date_trunc('month', block_time) as date) as block_month
+    , tx_id
+FROM {{ source('solana','instruction_calls') }}
+WHERE executing_account = 'PERPHjGBqRHArX4DySjwM6UJHiR3sWAatqfdBS2qQJu'
+AND bytearray_substring(data,1+8,8) = 0xf5715534d6bb9984 -- IncreasePosition
+AND tx_success = true
+{% if is_incremental() %}
+AND {{ incremental_predicate('block_time') }}
+{% endif %}
+
+UNION ALL 
+
+-- DecreasePositionEvent
+-- decrease has pnl and has profit indicators.
+-- example decrease https://solana.fm/tx/5PURicyWYuAiYsJHgzgubuBfsoSYXQYX812RhmdbyCUtLRrpd6WRq6GNcrhEGXnkB5hZ9zHr4LMiagQGrfPYnGYo?cluster=mainnet-qn1
+SELECT
+    'decrease' AS position_change
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+292,8))) / 1e6 as size_usd --this is the levered size delta (change)
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+300,8))) / 1e6 as collateral_usd --collateral size, transferred back to owner
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+309,8))) as collateral_token --collateral notional size, transferred back to owner
+    , case when bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+325,1))) = 1 
+        then bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+334,8))) / 1e6 
+        else bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+326,8))) / 1e6
+        end as fee_usd --close fee, has an optional param priceSlippage before it so we need this case when.
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+317,8))) / 1e6 as price_usd --exit price
+    , null as liq_fee_usd
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+219,1))) as pnl_direction --0 means negative, 1 means positive
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+220,8))) / 1e6 as pnl_usd --pnl only on decrease position and liquidation
+    , to_base58(bytearray_substring(data,1+228,32)) as owner
+    , to_base58(bytearray_substring(data,1+16,32)) as position_key
+    , bytearray_to_bigint(bytearray_reverse(bytearray_substring(data,1+16+32,1))) as position_side --1 is long, 2 is short
+    , to_base58(bytearray_substring(data,1+16+32+1,32)) as custody_position_key --ties to the token mint of long or short token
+    , to_base58(bytearray_substring(data,1+16+32+1+32,32)) as custody_collateral_key --ties to the token mint of the collateral token
+    , block_slot
+    , block_time
+    , date_trunc('month', block_time) as block_month
+    , tx_id
+FROM {{ source('solana','instruction_calls') }}
+WHERE executing_account = 'PERPHjGBqRHArX4DySjwM6UJHiR3sWAatqfdBS2qQJu'
+AND bytearray_substring(data,1+8,8) = 0x409c2b4a6d83107f -- DecreasePosition
+AND tx_success = true
+{% if is_incremental() %}
+AND block_time >= date_trunc('day', now() - interval '1' day)
```

Comment:
please update here too

Reply:
it is null only in 2021. Does not affect perps which were deployed in 2023